### PR TITLE
update bootstrap and default version to v0.4.0

### PR DESCRIPTION
**Pull request by dekkagaijin**
Signed-off-by: Jake Sanders <jsand@google.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,9 +21,9 @@ jobs:
     name: Install Cosign and test presence in path
     steps:
       - name: Install Cosign
-        uses: sigstore/cosign-installer@v0.1.0
+        uses: sigstore/cosign-installer@main
```
Comment (dekkagaijin) [inline README.md:24]: IMHO we should encourage keeping this as a floating, up-to-date reference

File: action.yml [hunk 2]
```diff
@@ -9,16 +9,16 @@ inputs:
...
   using: "composite"
   steps:
-    - run: curl -L https://storage.googleapis.com/cosign-releases/v0.2.0/cosign-linux-amd64 -o cosign
+    - run: curl -L https://storage.googleapis.com/cosign-releases/v0.4.0/cosign-linux-amd64 -o cosign
       shell: bash
     # We verify the version against a SHA **in the published action itself**, not in the GCS bucket.
     - run: |
         shaBootstrap=$(shasum -a 256 cosign | cut -d' ' -f1);
-        if [[ $shaBootstrap != "677f64870f209b1f41a02247a65697138973e3fb36a0836565cf173b9282aad6" ]]; then exit 1; fi
+        if [[ $shaBootstrap != "7425fb2cce21eb058e85d8a86bbb6d0e17ed7d98a25929f35835a6d910547e5e" ]]; then exit 1; fi
```
Comment (dekkagaijin) [inline action.yml:21]: https://github.com/sigstore/cosign/releases/download/v0.4.0/cosign-linux-amd64.sha256

File: action.yml [hunk 3]
```diff
@@ -9,16 +9,27 @@ inputs:
...
-    default: 'v0.2.0'
+    default: 'v0.4.0'
 runs:
   using: "composite"
   steps:
-    - run: curl -L https://storage.googleapis.com/cosign-releases/v0.2.0/cosign-linux-amd64 -o cosign
+    - run: |
+        sudo apt-get update -q
+        sudo apt-get install -yq \
+             libpcsclite1
```
Comment (mbestavros) [inline action.yml:19]: I can't see this used anywhere below. Why do we need this installed?

**Review by mbestavros — CHANGES_REQUESTED** [review]
Looks great for the most part. Just one small question to resolve, then I'll merge.

File: action.yml (hunk 3 above)
Comment (dekkagaijin) [reply action.yml:19]: it's used by cosign itself (for yubikey support). We do a similar step in `sigstore/cosign`: https://github.com/sigstore/cosign/blob/main/.github/workflows/build.yaml#L39-L40

**Review by mbestavros — APPROVED** [review]
LGTM!

**Pull request merged**